docs/api-guide.md

```diff
@@ -894,60 +944,59 @@ public class Sample {
+- If you catch `CommitConflictException`, this exception indicates that committing the transaction has failed due to transient faults (for example, a conflict error). In this case, you can retry the transaction from the beginning.
+- If you catch `UnknownTransactionStatusException`, this exception indicates that the status of the transaction, whether it was successful or not, is unknown. In this case, you need to check if the transaction is committed successfully and retry the transaction if it has failed.
 
-In the sample code, for `UnknownTransactionStatusException`, the transaction is not retried because the cause of the exception is nontransient.
-Also, for `UnsatisfiedConditionException`, the transaction is not retried because how to handle this exception depends on your application requirements.
-For other exceptions, the transaction is retried because the cause of the exception is transient or nontransient.
-If the cause of the exception is transient, the transaction may succeed if you retry it.
-However, if the cause of the exception is nontransient, the transaction may still fail even if you retry it.
-In such a case, you will exhaust the number of retries.
+How to identify a transaction status is delegated to users. You may want to create a transaction status table and update it transactionally with other application data so that you can get the status of a transaction from the status table.
 
-Please note that if you begin a transaction by specifying a transaction ID, you must use a different ID when you retry the transaction.
-And, in the sample code, the transaction is retried three times maximum and sleeps for 100 milliseconds before it is retried.
-But you can choose a retry policy, such as exponential backoff, according to your application requirements.
+### Notes about some exceptions
 
-## Transactional operations for Two-phase Commit Transaction
+Although not illustrated in the example code, the `resume()` API could also throw `TransactionNotFoundException`. This exception indicates that the transaction associated with the specified ID was not found and/or the transaction might have expired. In either case, you can retry the transaction from the beginning since the cause of this exception is basically transient.
 
-Please see [Two-phase Commit Transactions](two-phase-commit-transactions.md).
+In the sample code, for `UnknownTransactionStatusException`, the transaction is not retried because the cause of the exception is non-transient. For other exceptions, the transaction is retried because the cause of the exception is transient or non-transient. If the cause of the exception is transient, the transaction may succeed if you retry it. However, if the cause of the exception is non-transient, the transaction may still fail even if you retry it. In such a case, you will exhaust the number of retries.
```

Comment:
The reason why `UnknownTransactionStatusException` shouldn't be retried without checking if the transaction succeeded is to avoid potential duplicated operations (e.g., multiple records creation), IIUC. So, this description sounds a bit weird to me.  @brfrn169 What do you think?

Reply:
> The reason why UnknownTransactionStatusException shouldn't be retried without checking if the transaction succeeded by the application is to avoid potential duplicated operations (e.g., multiple records creation), IIUC.

This is correct.

However, I'm actually not sure if "non-transient" is the most appropriate term to use in this context.

@josh-wong What do you think?

Reply:
@komamitsu Thank you for the explanation!

@brfrn169 Thank you for confirming! I agree, I don't think "non-transient" is the right word here. How about if we change the first sentence as follows?:

```suggestion
In the sample code, for `UnknownTransactionStatusException`, the transaction is not retried because the application must check if the transaction was successful to avoid potential duplicate operations. For other exceptions, the transaction is retried because the cause of the exception is transient or non-transient. If the cause of the exception is transient, the transaction may succeed if you retry it. However, if the cause of the exception is non-transient, the transaction will still fail even if you retry it. In such a case, you will exhaust the number of retries.
```

Reply:
@josh-wong Sounds good. Thanks!

Reply:
@josh-wong Can you please apply the same change to the Two-phase commit transaction document?
https://github.com/scalar-labs/scalardb/pull/1082/files#diff-2677f880f1a9e512f31cce8e12950863ddffd4be49408d310d72d4d73e6122b9R697

Reply:
Done!

**Ref:** https://github.com/scalar-labs/scalardb/blob/334069fe6d64dfb5f4df4aa33ea7de94b28d8383/docs/two-phase-commit-transactions.md?plain=1#L697C4-L697C4